backend/app/routes/events.py

```
@@ -44,169 +46,137 @@ def get_language_from_header(accept_language: Optional[str] = Header(None)) -> s
-    ),
-    longitude: Optional[float] = Query(
-        None, description="Longitude for geographic search"
-    ),
-    radius_km: Optional[float] = Query(None, description="Search radius in kilometers"),
-    language: Optional[str] = Query(None, description="Language code for translations"),
-    page: int = Query(1, ge=1, description="Page number"),
-    size: int = Query(20, ge=1, le=100, description="Page size"),
-    use_cache: bool = Query(
-        True, description="Use cached results for better performance"
-    ),
+    search_params: EventSearchParams = Depends(),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)